californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -1023,20 +1030,26 @@ public CoapObserveRelation observe(CoapHandler handler, int accept) {
 	}
 
 	/**
-	 * Shutdown the client-specific executor service, when not detached. Only
-	 * needed if {@link #useExecutor()} or {@link #setExecutor(ExecutorService)}
-	 * are used (i.e., a client-specific executor service was set).
+	 * Shutdown the client-specific executor service, when not detached. Always
+	 * needed unless you used detached executor.
 	 */
 	public void shutdown() {
 		ExecutorService executor;
+		ExecutorService secondaryExecutor;
 		boolean shutdown;
 		synchronized (this) {
 			executor = this.executor;
+			secondaryExecutor = this.secondaryExecutor;
 			shutdown = !this.detachExecutor;
 			this.executor = null;
 		}
-		if (shutdown && executor != null) {
-			executor.shutdownNow();
+		if (shutdown) {
```

Comment:
Is it intended, not to reset the `this.secondaryExecutor = null`?

Reply:
I will fix this.